identity-iota/src/tangle/message_set.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::BTreeMap;
+// use identity_common::core::FromJson;
+
+use crate::did::DocumentDiff;
+use crate::did::IotaDID;
+use crate::did::IotaVerificationMethod;
+use crate::did::Verifier;
+use crate::tangle::Message;
+use crate::tangle::MessageId;
+use crate::tangle::MessageIndex;
+use crate::tangle::TangleRef;
+use crate::tangle::TryFromMessage;
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct MessageSet<T> {
+  data: BTreeMap<MessageId, T>,
+  spam: Option<Vec<MessageId>>,
+}
```

Comment:
Using `Option<Vec<_>>`  seems unnecessary (unless it has more meaning attached, see my comment on making retaining spam optional) as the size of `None` is the same as an empty `Vec`.

Same comment for `DiffSet`.